### ISSUE TITLE
Formula Row(A1) is not supported

### ISSUE THREAD

**Issue by MSBudak** [ClosedXML/ClosedXML#187]
I'm getting an syntax error when I'm trying to get the value of a calculated cell. I think formula ROW is not implemented yet. Is there any chance that you implement this function ?

**Comment by igitur** [comment]
No, `ROW` and functions which refer to the cell's own location isn't implemented yet. It's tricky.

**Comment by jaklithn** [comment]
We also needed the same function. We assumed it was easy to fix but maybe no one had time to do it. We planned to contribute but hesitaed when we read the comment above.
@igitur When you say "It's tricky" what is the reason for this? Do you have any suggestions on how to implement it?

**Comment by igitur** [comment]
When a formula is evaluated, it has no idea about the cell in which it executes. So a formula that implicitly references itself will not eg `ROW() ` will not know what to return.

**Comment by jaklithn** [comment]
I have now checked the code and understand what you mean. On the other hand I think the formula evaluation really should be cell aware! There are more functions than `Row()` that would need that.

We are trying to achieve the column range references that are possible to use in Excel. If you define column range `A:A` with name `Term1` and column range `B:B` with name `Term2` then you can in column C add this formula to all cells: `=Term1+Term2`. This works perfect in Excel so somehow it manages to keep track of row position and solve the range reference into the relevant row reference.

**Comment by igitur** [comment]
> 

> I have now checked the code and understand what you mean. On the other hand I think the formula evaluation really should be cell aware! There are more functions than `Row()` that would need that.

I agree.

> We are trying to achieve the column range references that are possible to use in Excel. If you define column range `A:A` with name `Term1` and column range `B:B` with name `Term2` then you can in column C add this formula to all cells: `=Term1+Term2`. This works perfect in Excel so somehow it manages to keep track of row position and solve the range reference into the relevant row reference.

Do you mean the `=[@Term1] + [@Term2]` style of referencing? If so, that's a separate issue to this issue and is also not implemented. 

As always, we welcome pull requests.

**Comment by jaklithn** [comment]
> Do you mean the =[@Term1] + [@Term2] style of referencing? 
No, not really ... I am not familiar with that syntax. I meant exactly the way I wrote it. But I still realize it calls for a major rewrite of the calculation evaluation. I tried to do it but lost track in the expression cache that I did't understand.

**Comment by StevenJonSmith** [comment]
Is there anyway to not do the calculation before saving?

I am being asked to write into a template that uses ROW() extensively, I tried setting all SaveOptions to false but it still evaluates and throws.

**Comment by jahav** [comment]
Duplicate of #1851.
